### PR TITLE
Bug 5831/v3

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5831

Previous PR: https://github.com/OISF/suricata/pull/9373

Changes since v2:
- fix commit check

### CODE REVIEW

File: src/source-af-packet.c [hunk 1]
```diff
@@ -527,12 +526,14 @@ static TmEcode AFPPeersListAdd(AFPThreadVars *ptv)
                 continue;
             peer->peer = pitem;
             pitem->peer = peer;
-            mtu = GetIfaceMTU(ptv->iface);
-            out_mtu = GetIfaceMTU(ptv->out_iface);
-            if (mtu != out_mtu) {
+
+            LiveDevice *iface = ptv->livedev;
+            LiveDevice *out_iface = LiveGetDevice(ptv->out_iface);
```
Comment (victorjulien) [inline src/source-af-packet.c:531]: looks risky, what if we get a NULL ptr?

File: src/suricata.c [hunk 2]
```diff
@@ -2496,7 +2496,8 @@ static int ConfigGetCaptureValue(SCInstance *suri)
                             dev[len-1] = '\0';
                         }
                     }
-                    unsigned int iface_max_packet_size = GetIfaceMaxPacketSize(dev);
+                    LiveDevice *ld = LiveGetDevice(dev);
```
Comment (victorjulien) [inline src/suricata.c:2499]: can we get a NULL result here?

File: src/util-ioctl.c [hunk 3]
```diff
@@ -124,21 +114,23 @@ int GetIfaceMTU(const char *pcap_dev)
...
- * \param Name of a network interface
+ * \param LiveDevice object
  * \retval 0 in case of error
  */
-int GetIfaceMaxPacketSize(const char *pcap_dev)
+int GetIfaceMaxPacketSize(LiveDevice *ld)
 {
-    if ((pcap_dev == NULL) || strlen(pcap_dev) == 0)
+    const char *dev = ld->dev;
+    if ((dev == NULL) || strlen(dev) == 0)
```
Comment (victorjulien) [inline src/util-ioctl.c:123]: can `ld` itself be NULL?

**Review by victorjulien — CHANGES_REQUESTED** [review]

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15577

File: src/suricata.c (hunk 2 above)
Comment (inashivb) [reply src/suricata.c:2499]: yes.

File: src/source-af-packet.c (hunk 1 above)
Comment (inashivb) [reply src/source-af-packet.c:531]: indeed it is. I should check better. Sending new PR

**Pull request closed without merging**